libtransact/src/protocol/batch.rs

```diff
@@ -141,6 +141,61 @@ impl Batch {
     }
 }
 
+impl FromProto<protos::batch::Batch> for Batch {
+    fn from_proto(batch: protos::batch::Batch) -> Result<Self, ProtoConversionError> {
+        Ok(Batch {
+            header: batch.header,
+            header_signature: batch.header_signature,
+            transactions: batch
+                .transactions
+                .into_iter()
+                .map(|txn| txn.into_native())
+                .collect::<Result<Vec<_>, _>>()?,
+            trace: batch.trace,
+        })
+    }
+}
+
+impl FromNative<Batch> for protos::batch::Batch {
+    fn from_native(batch: Batch) -> Result<Self, ProtoConversionError> {
+        let mut proto_batch = protos::batch::Batch::new();
+        proto_batch.set_header(batch.header);
+        proto_batch.set_header_signature(batch.header_signature);
+        proto_batch.set_transactions(
+            batch
+                .transactions
+                .into_iter()
+                .map(|txn| txn.into_proto())
+                .collect::<Result<Vec<_>, _>>()?
+                .into(),
+        );
+        proto_batch.set_trace(batch.trace);
+        Ok(proto_batch)
+    }
+}
+
+impl FromBytes<Batch> for Batch {
+    fn from_bytes(bytes: &[u8]) -> Result<Batch, ProtoConversionError> {
+        let proto: protos::batch::Batch = protobuf::parse_from_bytes(bytes).map_err(|_| {
+            ProtoConversionError::SerializationError("Unable to get Batch from bytes".to_string())
```

Comment:
This should probably include the error string from the causing error. "Unable to get Batch from bytes: {}"

Reply:
Added commit to update the error messages